test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -2460,6 +2460,201 @@ static public void testFlattenedFieldInitSequence() throws Throwable {
+		String fields2[] = {};
+		Class usingClass = ValueTypeGenerator.generateRefClass("UsingUnresolvedA", fields2, "UnresolvedA");
+
+		MethodHandle defaultValueUnresolved = lookup.findStatic(usingClass, "testUnresolvedValueTypeDefaultValue", MethodType.methodType(Object.class, int.class));
+
+		for (int i = 0; i < 10; i++) {
+			// Pass zero to avoid execution of DEFAULTVALUE and resolution of value type class
+			assertNull(defaultValueUnresolved.invoke(0));
+		}
+
+		MethodHandle getX = generateGenericGetter(valueClass, "x");
+		MethodHandle getY = generateGenericGetter(valueClass, "y");
+
+		for (int i = 0; i < 10; i++) {
+			// Pass one to force execution of DEFAULTVALUE and resolution of value type class
+			Object defaultValue = defaultValueUnresolved.invoke(1);
+			assertNotNull(defaultValue);
+			assertEquals(getX.invoke(defaultValue), Integer.valueOf(0));
+			assertEquals(getY.invoke(defaultValue), Integer.valueOf(0));
+		}
+	}
+
+	@Test(priority=1)
+	static public void testUnresolvedWithFieldUse() throws Throwable {
+		String fields[] = {"x:I", "y:I"};
+		Class valueClass = ValueTypeGenerator.generateValueClass("UnresolvedD", fields, "HostD");
+		String fields2[] = {};
+		Class usingClass = ValueTypeGenerator.generateHostRefClass("UsingUnresolvedD", fields2, "UnresolvedD", fields);
+
+		MethodHandle withFieldUnresolved = lookup.findStatic(usingClass, "testUnresolvedValueTypeWithField", MethodType.methodType(Object.class, new Class[] {int.class, Object.class}));
+
+		for (int i = 0; i < 10; i++) {
+			// Pass zero to avoid execution of WITHFIELD and resolution of value type class
+			assertNull(withFieldUnresolved.invoke(0, null));
+		}
+
+		MethodHandle makeDefaultValue = lookup.findStatic(valueClass, "makeValueTypeDefaultValue", MethodType.methodType(Object.class));
+		Object defaultValue = makeDefaultValue.invoke();
+
+		MethodHandle getX = generateGenericGetter(valueClass, "x");
+		MethodHandle getY = generateGenericGetter(valueClass, "y");
+
+		for (int i = 0; i < 10; i++) {
+			// Pass one to force execution of WITHFIELD and resolution of value type class
+			Object withFieldValue = (withFieldUnresolved.invoke(1, defaultValue));
+			assertEquals(getX.invoke(withFieldValue), Integer.valueOf(1));
+			assertEquals(getY.invoke(withFieldValue), Integer.valueOf(2));
+		}
+	}
+
+	private static class UnresolvedClassDesc {
+		public String name;
+		public String[] fields;
+		public UnresolvedClassDesc(String name, String[] fields) {
+			this.name = name;
+			this.fields = fields;
+		}
+	}
+
+	/*
+	 * Test use of GETFIELD operations on the fields of a container class, where
+	 * the fields are of value type classes that have not been resolved.
+	 *
+	 * The method is first called so that the GETFIELD will not be executed, and
+	 * the class not resolved, and then called so that the GETFIELD and class
+	 * resolution is triggered.
+	 */
+	@Test(priority=1)
+	static public void testUnresolvedGetFieldUse() throws Throwable {
```

Comment:
Would adding some java example code on how `UnresolvedB*` , `ContainerForUnresolvedB`, `UsingUnresolvedB`, and getters are defined and used helps people follow the test better? Same question on `testUnresolvedPutFieldUse`.

Reply:
Yes, definitely.  I'll do that.

Reply:
Annabelle @a7ehuo, I have addressed this in commit 57a652d00483cac5c92f341e93509a40bd5516f2.  May I ask you to have a look?